framework/decode/vulkan_address_replacer.cpp

```diff
@@ -0,0 +1,791 @@
+                        addresses_to_replace.push_back(accel_structure_reference);
+                    }
+                    break;
+                }
+                default:
+                    GFXRECON_LOG_ERROR(
+                        "OverrideCmdBuildAccelerationStructuresKHR: unhandled case in switch-statement: %d",
+                        geometry->geometryType);
+                    break;
+            }
+        }
+    }
+
+    if (!addresses_to_replace.empty())
+    {
+        // prepare linear hashmap
+        hashmap_bda_.clear();
+        auto acceleration_structure_map = address_tracker.GetAccelerationStructureDeviceAddressMap();
+        for (const auto& [capture_address, replay_address] : acceleration_structure_map)
+        {
+            if (force_replace || capture_address != replay_address)
+            {
+                // store addresses we will need to replace
+                hashmap_bda_.put(capture_address, replay_address);
+            }
+        }
+
+        if (!hashmap_bda_.empty())
+        {
+            // mark injected commands
+            decode::BeginInjectedCommands();
+
+            if (pipeline_bda_ == VK_NULL_HANDLE)
+            {
+                init_pipeline();
+            }
+
+            if (!create_buffer(hashmap_bda_.get_storage(nullptr), pipeline_context_bda_.hashmap_storage))
+            {
+                GFXRECON_LOG_ERROR("VulkanAddressReplacer: hashmap-storage-buffer creation failed");
+            }
+            hashmap_bda_.get_storage(pipeline_context_bda_.hashmap_storage.mapped_data);
+
+            uint32_t num_bytes = addresses_to_replace.size() * sizeof(VkDeviceAddress);
+
+            if (!create_buffer(num_bytes, pipeline_context_bda_.input_handle_buffer))
+            {
+                GFXRECON_LOG_ERROR("VulkanAddressReplacer: input-handle-buffer creation failed");
+            }
+            memcpy(pipeline_context_bda_.input_handle_buffer.mapped_data, addresses_to_replace.data(), num_bytes);
+
+            replacer_params_t replacer_params = {};
+            replacer_params.hashmap.storage   = pipeline_context_bda_.hashmap_storage.device_address;
+            replacer_params.hashmap.size      = hashmap_bda_.size();
+            replacer_params.hashmap.capacity  = hashmap_bda_.capacity();
+
+            // in-place
+            replacer_params.input_handles  = pipeline_context_bda_.input_handle_buffer.device_address;
+            replacer_params.output_handles = pipeline_context_bda_.input_handle_buffer.device_address;
+
+            replacer_params.num_handles = addresses_to_replace.size();
+
+            device_table_->CmdBindPipeline(command_buffer_info->handle, VK_PIPELINE_BIND_POINT_COMPUTE, pipeline_bda_);
+
+            // NOTE: using push-constants here requires us to re-establish the previous data, if any
+            device_table_->CmdPushConstants(command_buffer_info->handle,
```

Comment:
Be careful of push constants size.  Some HW (Arm?) may have no support or very limited support.

Reply:
glad you're asking :) 

- "Push constants have a minimum size of 128 bytes" (https://vkguide.dev/docs/chapter-3/push_constants/)
- `replacer_params_t` is only 40 bytes, so plenty of headroom